### PR TITLE
Test-Proxy Transition Report Script

### PR DESCRIPTION
**Pull request by scbedd**
See resulting markdown [on our wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/834/Test-Proxy-Transition-Overview) by clicking on the links for each language.

- [x] [summary report](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/834/Test-Proxy-Transition-Overview)
- [x] [go](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/841/Go-Details)
- [x] [python](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/836/Python-Details)
- [x] [js](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/839/JS-Details)
- [x] [.net](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/838/Net-Details)
- [x] [cpp](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/848/CPP-Details)
- ~~java?~~

### CODE REVIEW

File: scripts/python/assets-automation/generate_assets_report.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      Generates a markdown report that summarizes the the status of the transition to the test-proxy and externalized assets.
+      """
+    )
+    parser.parse_args()
+
+    # python = generate_python_report()
+    # write_output(python)
+
+    # js = generate_js_report()
+    # write_output(js)
```
Comment (scbedd) [inline scripts/python/assets-automation/generate_assets_report.py:287]: Once one of these is completed I comment it to increase dev-cycle speed. Uncomment when finished.

File: scripts/python/assets-automation/generate_assets_report.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+    language = "net"
+    result = ScanResult("." + language.upper())
+    repo = get_repo(language)
+
+    print(f"Evaluating repo for {language} @ {repo}", end="...")
+
+    #                                                     <service>
+    #                                                         |<package>
+    #                                                         |    |
+    all_azure_projects = glob.glob(os.path.join(repo, "sdk", "*", "*", "src", "*Azure.*.csproj"), recursive=True)
```
Comment (weshaggard) [inline scripts/python/assets-automation/generate_assets_report.py:248]: I assume the exact counts aren't critical for this report but if we did need exact counts we should try to use the Get-AllPackageInfoFromRepo functions we have in each language-settings.ps1 to discover the packages.

File: scripts/python/assets-automation/generate_assets_report.py [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+    return any([input_dir in subdir for subdir in directory_patterns])
+
+
+def js_package_included(package_path: str) -> bool:
+    package_name = os.path.basename(os.path.dirname(package_path))
+
+    excluded_packages = [
+        "samples-react",
+        "sample-react",
+        "mock-hub" "abort-controller",
```
Comment (HarshaNalluru) [inline scripts/python/assets-automation/generate_assets_report.py:350]:
```suggestion
        "mock-hub", "abort-controller",
```

File: scripts/python/assets-automation/generate_assets_report.py [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+        "logger",
+        "samples-express",
+        "samples-browser" "samples-react",
```
Comment (HarshaNalluru) [inline scripts/python/assets-automation/generate_assets_report.py:353]:
```suggestion
        "samples-browser", "samples-react",
```

File: scripts/python/assets-automation/generate_assets_report.py [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+    ]
+
+    excluded_package_postfixes = ["-track-1", "-common"]
+
+    excluded_package_prefixes = ["@azure/core-"]
+
+    # exclude any packages that have these paths in them
+    excluded_directories = [
+        os.path.join("sdk", "identity", "identity", "test"),
+        os.path.join("sdk", "test-utils"),
```
Comment (HarshaNalluru) [inline scripts/python/assets-automation/generate_assets_report.py:370]:
```suggestion
        os.path.join("sdk", "identity", "identity", "test"),
        os.path.join("sdk", "test-utils"),
        os.path.join("sdk", "core"),
```

File: scripts/python/assets-automation/generate_assets_report.py (hunk 2 above)
Comment (scbedd) [reply scripts/python/assets-automation/generate_assets_report.py:248]: I see where you're coming from but given how little of this script is actually _discovering_ the package, I don't necessarily see a point in debugging getting that working.

File: scripts/python/assets-automation/generate_assets_report.py [hunk 6]
```diff
@@ -0,0 +1,535 @@
...
+        "logger",
+        "samples-express",
+        "samples-browser", "samples-react",
+        "event-hubs-track-1",
+        "opentelemetry-instrumentation-azure-sdk",
+        "monitor-opentelemetry-exporter",
+        "service-bus-v1",
+        "service-bus-v7",
+        "app",
+        "perf",
```
Comment (HarshaNalluru) [inline scripts/python/assets-automation/generate_assets_report.py:375]: The ones that depend on AMQP, are to be skipped
	- service-bus
	- eventhubs-checkpointstore-blob
	- eventhubs-checkpointstore-tables
	- schema-registry-avro	

Nothing to record, so to be skipped
	- api-management-custom-widgets-scaffolder
	- storage-internal-avro
        - web-pubsub-express

Comment (HarshaNalluru) [reply scripts/python/assets-automation/generate_assets_report.py:375]:
```suggestion
        "perf",
        "service-bus",
        "eventhubs-checkpointstore-blob",
        "eventhubs-checkpointstore-tables",
        "schema-registry-avro",
        "api-management-custom-widgets-scaffolder",
        "storage-internal-avro",
        "web-pubsub-express",
```

**Review by weshaggard — APPROVED** [review]

**Pull request merged**